src/legacy/server/logging/rotate/log_rotator.ts

```diff
@@ -150,12 +152,20 @@ export class LogRotator {
   }
 
   async _startLogFileSizeMonitor() {
-    this.usePolling = await this._shouldUsePolling();
+    this.usePolling = this.config.get('logging.rotate.usePolling');
+    this.shouldUsePolling = await this._shouldUsePolling();
 
-    if (this.usePolling && this.usePolling !== this.config.get('logging.rotate.usePolling')) {
+    if (this.usePolling && !this.shouldUsePolling) {
       this.log(
         ['warning', 'logging:rotate'],
-        'The current environment does not support `fs.watch`. Falling back to polling using `fs.watchFile`'
+        'Looks like your current environment support `fs.watch` but you are using polling which uses `fs.watchFile`'
+      );
+    }
+
+    if (!this.usePolling && this.shouldUsePolling) {
+      this.log(
+        ['warning', 'logging:rotate'],
+        'The current environment does not support `fs.watch`. Please enable `usePolling` in order to use `fs.watchFile`'
```

Comment:
in this case log rotation doesn't work? should it be an error?

Reply:
the test is not 100% accurate has it can throws false positives for example if you ran out of available watchers. But anyway, I think it is a good idea to log an error :)